drivers/gnss/gnss_u_blox_m10.c

```diff
@@ -0,0 +1,888 @@
+
+	/* UART backend */
+	struct modem_pipe *uart_pipe;
+	struct modem_backend_uart uart_backend;
+	uint8_t uart_backend_receive_buf[UART_RECV_BUF_SZ];
+	uint8_t uart_backend_transmit_buf[UART_TRNF_BUF_SZ];
+
+	/* Modem chat */
+	struct modem_chat chat;
+	uint8_t chat_receive_buf[CHAT_RECV_BUF_SZ];
+	uint8_t *chat_argv[CHAT_ARGV_SZ];
+
+	/* Modem ubx */
+	struct modem_ubx ubx;
+	uint8_t ubx_receive_buf[UBX_RECV_BUF_SZ];
+	uint8_t ubx_work_buf[UBX_WORK_BUF_SZ];
+
+	/* Ubx frame */
+	uint8_t ubx_frame_buf[UBX_FRM_BUF_SZ];
+
+	struct k_spinlock lock;
+};
+
+MODEM_CHAT_MATCHES_DEFINE(unsol_matches,
+	MODEM_CHAT_MATCH_WILDCARD("$??GGA,", ",*", gnss_nmea0183_match_gga_callback),
+	MODEM_CHAT_MATCH_WILDCARD("$??RMC,", ",*", gnss_nmea0183_match_rmc_callback),
+#if CONFIG_GNSS_SATELLITES
+	MODEM_CHAT_MATCH_WILDCARD("$??GSV,", ",*", gnss_nmea0183_match_gsv_callback),
+#endif
+);
+
+static int ubx_m10_resume(const struct device *dev)
+{
+	struct ubx_m10_data *data = dev->data;
+	int ret;
+
+	ret = modem_pipe_open(data->uart_pipe);
+	if (ret < 0) {
+		return ret;
+	}
+
+	ret = modem_chat_attach(&data->chat, data->uart_pipe);
+	if (ret < 0) {
+		modem_pipe_close(data->uart_pipe);
+		return ret;
+	}
+
+	return ret;
+}
+
+static int ubx_m10_turn_off(const struct device *dev)
+{
+	struct ubx_m10_data *data = dev->data;
+
+	return modem_pipe_close(data->uart_pipe);
+}
+
+static int ubx_m10_init_nmea0183_match(const struct device *dev)
+{
+	struct ubx_m10_data *data = dev->data;
+
+	const struct gnss_nmea0183_match_config match_config = {
+		.gnss = dev,
+#if CONFIG_GNSS_SATELLITES
+		.satellites = data->satellites,
+		.satellites_size = ARRAY_SIZE(data->satellites),
+#endif
+	};
+
+	return gnss_nmea0183_match_init(&data->match_data, &match_config);
+}
+
+static void ubx_m10_init_pipe(const struct device *dev)
+{
+	const struct ubx_m10_config *cfg = dev->config;
+	struct ubx_m10_data *data = dev->data;
+
+	const struct modem_backend_uart_config uart_backend_config = {
+		.uart = cfg->uart,
+		.receive_buf = data->uart_backend_receive_buf,
+		.receive_buf_size = sizeof(data->uart_backend_receive_buf),
+		.transmit_buf = data->uart_backend_transmit_buf,
+		.transmit_buf_size = ARRAY_SIZE(data->uart_backend_transmit_buf),
+	};
+
+	data->uart_pipe = modem_backend_uart_init(&data->uart_backend, &uart_backend_config);
+}
+
+static uint8_t ubx_m10_char_delimiter[] = {'\r', '\n'};
+
+static int ubx_m10_init_chat(const struct device *dev)
+{
+	struct ubx_m10_data *data = dev->data;
+
+	const struct modem_chat_config chat_config = {
+		.user_data = data,
+		.receive_buf = data->chat_receive_buf,
+		.receive_buf_size = sizeof(data->chat_receive_buf),
+		.delimiter = ubx_m10_char_delimiter,
+		.delimiter_size = ARRAY_SIZE(ubx_m10_char_delimiter),
+		.filter = NULL,
+		.filter_size = 0,
+		.argv = data->chat_argv,
+		.argv_size = ARRAY_SIZE(data->chat_argv),
+		.unsol_matches = unsol_matches,
+		.unsol_matches_size = ARRAY_SIZE(unsol_matches),
+		.process_timeout = K_MSEC(2),
```

Comment:
The process_timeout no longer exists, please rebase on the latest main and it will be gone :)